google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreOptions.java

```diff
@@ -216,8 +222,69 @@ public FirestoreOptions build() {
         }
       }
 
+      // Override credentials and channel provider if we are using the emulator.
+      String emulatorHost = System.getenv(FIRESTORE_EMULATOR_SYSTEM_VARIABLE);
+      if (emulatorHost != null) {
+        String hostUrlString = "http://" + emulatorHost;
```

Comment:
Nit: Move this into the try/catch block to make sure this is never used anywhere else, especially not in `setEndpoint()`.